extensions/google-drive/src/db.ts

```diff
@@ -243,17 +243,14 @@ export const indexFiles = async (db: Database, options: IndexFilesOptions = { fo
       }`,
       message: "This may take some time, please wait...",
     });
-    let favoriteFilePaths: Array<string> = [];
 
-    if (options.force) {
-      // Backup the favorite file paths before force indexing
-      favoriteFilePaths = queryFavoriteFiles(db, 1000).map((file) => file.path);
+    // Backup the favorite file paths before indexing
+    const favoriteFilePaths = queryFavoriteFiles(db, 1000).map((file) => file.path);
 
-      // Delete all the old indexed files
-      db.exec("DELETE from files");
+    // Delete all the old indexed files
+    db.exec("DELETE from files");
```

Comment:
So that there's no difference between forced indexing and automatic indexing (on cache expiry). This will ensure that the no longer existing paths will be nuked off from the database (due to which the mentioned bug seemed to have occurred).